core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -103,9 +103,7 @@ class ZooKeeperClient(connectString: String,
   // Fail-fast if there's an error during construction (so don't call initialize, which retries forever)
   @volatile private var zooKeeper = new ZooKeeper(connectString, sessionTimeoutMs, ZooKeeperClientWatcher)
 
-  newGauge("SessionState", new Gauge[String] {
-    override def value: String = Option(connectionState.toString).getOrElse("DISCONNECTED")
-  })
+  newGauge("SessionState", () => connectionState.toString)
```

Comment:
The `Option` wrapping doesn't seem to achieve anything here. If `connectionState` is `null`, we'd get a NPE. `toString` should always return a non null value.